hudi-spark-datasource/hudi-spark3.2.x/src/main/scala/org/apache/spark/sql/adapter/Spark3_2Adapter.scala

```diff
@@ -123,8 +123,4 @@ class Spark3_2Adapter extends BaseSpark3Adapter {
     case OFF_HEAP => "OFF_HEAP"
     case _ => throw new IllegalArgumentException(s"Invalid StorageLevel: $level")
   }
-
-  override def failAnalysisForMIT(a: Attribute, cols: String): Unit = {
```

Comment:
This should be moved to `HoodieSpark32CatalystPlanUtils` correct?

Reply:
Do we have a unit test for the failed analysis?  A unit test should fail if this is missing for Spark 3.2.  If not, could we add one?

Reply:
I am seeing this compile error when moving it to `HoodieSpark32CatalystPlanUtils`

```
% mvn clean -T 12C package -DskipTests -Dspark3.2 -Dscala-2.12
[ERROR] /Users/amrish/code/amrish-hudi-1/hudi-spark-datasource/hudi-spark3.2.x/src/main/scala/org/apache/spark/sql/HoodieSpark32CatalystPlanUtils.scala:77: error: value failAnalysis is not a member of org.apache.spark.sql.catalyst.expressions.Attribute
[ERROR]     a.failAnalysis(s"cannot resolve ${a.sql} in MERGE command given columns [$cols]")
[ERROR] 
```

Reply:
Fixed. Add failAnalysisForMIT function to HoodieSpark[32,33]CatalystPlanUtils.scala